framework/main.py

```diff
@@ -10,77 +10,122 @@
 import sys
 
 
-def init_argparser():
+def init_argparser(args=None):
     """
     Initialization  of argument parse library with it's arguments
     Args:
-        None
+        args: list of args to parse in `sys.argv` form. Useful for testing
+            passed to `ArgumentParser.parse_args` . default to None (use `sys.argv`)
 
     Returns:
         object: Parsed arguments - returned from parser.parse_args()
     """
+    server_args_parser = ArgumentParser(add_help=False)
```

Comment:
i think these arguments since they're shared by all of the sub-commands and relate to the generic configuration of the server, they should just be added to the `parser` instead of sharing them via `parents` to all the individual sub-parsers.

I think that results in the difference of:
```
./my_cli --server-opt  ACTION --action-opt
```

vs.

```
./my_cli ACTION --server-opt --action-opt
```

but i'm not sure... i haven't really used this pattern before. Based on the linked stackoverflow questions I think the `parents` way of sharing options is meant for sub-commands where some share the options and some don't, these options however look like they're always optional for all sub-commands...

though i guess that's personal preference...

Reply:
Yeah I went back and forth a few times on this, I think i ran into an issue adding them to `parser` and decided to switch back to the way it is now. 

I my opinion though things like `--server` and `--token` really should be coming from some sort of config file so you dont have to type them everytime. We still want them to be options but I think it gets kinda unwieldy having them be part of the command line all the time.